spec/mongo/client_construction_spec.rb

```diff
@@ -155,18 +155,42 @@
 
     context 'when providing options' do
 
-      context 'when retry_writes is defined' do
+      context 'retry_writes option' do
+        let(:client) do
+          new_local_client_nmio(SpecConfig.instance.addresses, options)
+        end
 
-        let(:options) do
-          { retry_writes: true }
+        context 'when retry_writes is true' do
+
+          let(:options) do
+            { retry_writes: true }
+          end
+
+          it 'sets retry_writes to true' do
+            expect(client.options['retry_writes']).to be true
+          end
         end
 
-        let(:client) do
-          new_local_client_nmio(SpecConfig.instance.addresses, authorized_client.options.merge(options))
+        context 'when retry_writes is false' do
+
+          let(:options) do
+            { retry_writes: false }
+          end
+
+          it 'sets retry_writes to false' do
+            expect(client.options['retry_writes']).to be false
+          end
         end
 
-        it 'sets the option' do
-          expect(client.options['retry_writes']).to eq(options[:retry_writes])
+        context 'when retry_writes is not given' do
+
+          let(:options) do
+            { retry_writes: true }
```

Comment:
This test looks identical to `when retry_writes is true`. I think it should set the options to the empty hash?

Reply:
Good catch, fixed.